app/models/vim_performance_state.rb

```diff
@@ -231,6 +221,25 @@ def capture_assoc_ids
     self.assoc_ids = result.presence
   end
 
+  def fetch_assoc_records(resource, assoc)
+    method = if assoc == :vms
+               if resource.kind_of?(EmsCluster)
+                 :all_vms_and_templates
+               elsif resource.kind_of?(Service)
+                 :vms
+               else
+                 :vms_and_templates
+               end
+             else
+               assoc
+             end
+    return false unless resource.respond_to?(method)
+
+    records = resource.send(method)
+    records = records.not_archived_before(timestamp) if records.try(:klass).respond_to?(:not_archived_before)
```

Comment:
This is the only actual changed line

Reply:
> Why do we have the `try` here?
> -- @Fryguy 

If the association is a virtual attribute, like it is for `MiqRegion`, then it will return an array that does not have a `klass`.